repositories/src/main/scala/cromwell/pipeline/datastorage/dto/ProjectConfiguration.scala

```diff
@@ -6,16 +6,24 @@ import ProjectFile.pathFormat
 import org.mongodb.scala.bson.collection.immutable.Document
 import play.api.libs.json.{ Json, OFormat }
 
-case class ProjectFileConfiguration(path: Path, inputs: List[FileParameter])
+case class ProjectFileConfiguration(path: Path, inputs: Set[FileParameter])
 
 object ProjectFileConfiguration {
   implicit val projectFileConfigurationFormat: OFormat[ProjectFileConfiguration] = Json.format
+
+  def toMap(config: ProjectFileConfiguration): Map[Path, Map[String, TypedValue]] =
+    Map(
+      config.path -> config.inputs.flatMap(fileParameter => Map(fileParameter.name -> fileParameter.typedValue)).toMap
+    )
+
+  def fromMap(path: Path, inputs: Map[String, TypedValue]): ProjectFileConfiguration =
+    ProjectFileConfiguration(path, inputs.flatMap(x => Set(FileParameter(x._1, x._2))).toSet)
 }
 
 case class ProjectConfiguration(
   projectId: ProjectId,
   active: Boolean,
-  projectFileConfigurations: List[ProjectFileConfiguration]
+  projectFileConfigurations: Set[ProjectFileConfiguration]
```

Comment:
`Set` here is rather useless. It won't save us from cases when we have two configurations with equal paths but different inputs.
If you want to make sure there is no duplicate configurations then let's use `Map` directly.
Something like 
```suggestion
  configurations: Map[Path, Configuration]
```
where Configuration is
```
case class Configuration(inputs: Set[FileParameter])
```
should be good enough.
Or we can leave it as is. I just don't think using `Set` here is a good idea because it gives a false perception that these configurations would be unique the way we want them to be.

Reply:
I faced problems with marshaling/unmarshaling when I used `configurations: Map[Path, Configuration]`. We need to use List, not Map.
So I decided to leave it as is.

Reply:
If you faced some problems then why didn't you ask how to solve them? :)
Take a look at `play.api.libs.json.DefaultWrites#keyMapWrites` and `play.api.libs.json.DefaultReads#mapReads`